server/app/graph.go

```diff
@@ -26,3 +29,42 @@ func (a *App) GetFrontEndGraph() *model.FrontendGraph {
 
 	return &gr
 }
+
+func (a *App) GetNextNodes(userID string) ([]model.Node, []model.Node, error) {
+	statuses, err := a.GetStatusesForUser(userID)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "can't get statuses for user %s", userID)
+	}
+	statusMap := map[string]*model.NodeStatusForUser{}
+	for _, status := range statuses {
+		statusMap[status.NodeID] = status
+	}
+
+	inProgressNodes := []model.Node{}
+	nextNodes := []model.Node{}
+
+	for _, node := range a.Graph.Nodes {
+		status, ok := statusMap[node.ID]
+		if ok && (status.Status == model.NodeStatusStarted || status.Status == model.NodeStatusWatched) {
+			inProgressNodes = append(inProgressNodes, node)
+		} else if !ok || status.Status != model.NodeStatusFinished {
+			prereqs, ok2 := a.Graph.Prerequisites[node.ID]
+			if !ok2 {
+				return nil, nil, errors.Wrapf(err, "no prerequisites for node %s", node.ID)
+			}
```

Comment:
Why do we return an empty list when one of the nodes doesn't have prerequisites?